src/discordtipbot/tipbot.cr

```diff
@@ -229,6 +229,10 @@ class TipBot
     end
   end
 
+  def get_high_balance(high_balance : Int32)
+    @db.query_all("SELECT userid FROM accounts WHERE balance > $1", high_balance, &.read(Int64 | Nil))
```

Comment:
Why `Int64 | Nil`? `userid` is not a nullable column, and the block should not be run if the result set is empty. Seems like you should just be using [this overload](http://crystal-lang.github.io/crystal-db/api/0.5.0/DB/QueryMethods.html#query_all%28query%2C%2Aargs%2Castype%3AClass%29-instance-method), which should return `Array(UInt64)` in your case.

Reply:
Yeah, looking at the [base overload implementation](https://github.com/crystal-lang/crystal-db/blob/be139c900c86e28b150b9808c74d2a4dd2bb913b/src/db/query_methods.cr#L194-L206) for `query_all`, it should always return an array, not `Array(U) | Nil`:
```cr
    def query_all(query, *args, &block : ResultSet -> U) : Array(U) forall U
      ary = [] of U
      query_each(query, *args) do |rs|
        ary.push(yield rs)
      end
      ary
    end
```

Reply:
Cool. Somehow I missed that overload.